Sortbyexpdate2

miralsumrani opened the pull request:
Sort function for expiry date

Alikpld committed:
Connected the search function to the main page

the "test" search function is now working on the main page, we can get rid of the /db page. Next step is to integate search with the add product

miralsumrani committed:
commit

miralsumrani committed:
updated sort function

miralsumrani committed:
Updated sort by expiry date function

miralsumrani committed:
Updated sort by expiry date function + css formatting

miralsumrani committed:
added 'user_ID = 1' to queries

miralsumrani committed:
Disclaimer added on index + addProducts page for expiry dates

nandatanvi committed:
Added Sign In Page

nmdoran committed:
Add files via upload

nmdoran committed:
Update client.js

nmdoran committed:
Update index.js

Commit:
Matthew to Check

mdschnei committed:
change 'product' to 'product_name' in query

mishrapalak committed:
basic-email-functionality

mdschnei committed:
fixed comment syntax that was stopping client.js page, hard coded search-add quantity to 1

mishrapalak committed:
Merge pull request #21 from nmdoran/SignInPage

Sign In Page Created

miralsumrani committed:
EditQuantity commit

nandatanvi committed:
testing trigger

miralsumrani committed:
update to query

nandatanvi committed:
Merge pull request #22 from nmdoran/email-functionality

basic-email-functionality

nandatanvi committed:
Added Email Trigger

mdschnei committed:
change logic order of edit quantity query

miralsumrani committed:
commit

nandatanvi committed:
Added Trigger for Email

mishrapalak commented:
Hi Miral, this is working on my computer, great job!

mishrapalak committed:
Merge pull request #24 from nmdoran/email-functionality-trigger

Email functionality trigger

miralsumrani committed:
Merge branch 'editQuantity' of https://github.com/nmdoran/MSCI342 into editQuantity

miralsumrani committed:
changing HTML id to quantity2

miralsumrani committed:
keeping query quantity the same

mdschnei committed:
added back missing closing bracket

mdschnei committed:
remove console log oops

Commit:
Merge https://github.com/nmdoran/MSCI342 into Integrate-search-with-add-product

Commit:
Merge branch 'Integrate-search-with-add-product' of https://github.com/nmdoran/MSCI342 into Integrate-search-with-add-product

Commit:
Filter by Food Type

The user story filter by food type is completed. Once you select something from the dropdown it should filter the table and everything in it to adjust the filter.

nmdoran committed:
Updated Database

1) Changed datatype char to varchar(255) in all instances except postal code
2) Added autoincrementing ID for products table
3) Added user_ID column to products table.  The default value is NULL which means any user can access.

miralsumrani committed:
Update editQuantity.ejs

miralsumrani committed:
Update index.ejs

nmdoran committed:
Update index.js

mdschnei commented on index.js:
```diff
@@ -19,16 +19,28 @@ express()
+  FROM fridge_products f
+  INNER JOIN products p ON f.prod_ID = p.prod_ID
+  WHERE user_ID = '1'
+  ORDER BY exp_dt ${req.query.expirysort}`)
```
Due to changes that we've just made to the layout of the product table, whereas this query before was perfectly fine, it now requires just one more tiny fit of specificity, in that the user_ID = '1' should be amended to f.user_ID = '1', since there is now also a user_ID column in the products table and the query currently doesn't know which table the where statement is referring to.

nmdoran committed:
Fixed spelling of product, minor query fix

mdschnei commented on public/client.js:
```diff
@@ -30,4 +30,10 @@ function removeProduct() {
+
+function sortbyExpiry() {
+  var x = document.getElementById("expirysort").value;
+  window.location.replace("http://localhost:5000/?expirysort="+x);
```
Sorry, I realized I missed something here earlier with this, the problem being that this code will always attempt to redirect to localhost:5000, even when it is being run on the live site hosted by Heroku, which we obviously don't want. To fix this for now, please add the following to make the page redirect accordingly:
```
if (window.location.hostname === "localhost") {
    window.location.replace("http://localhost:5000/?expirysort="+x);
  } else {
    window.location.replace("https://whatscookinggoodlooking.herokuapp.com/?expirysort="+x);
  }
```

nandatanvi committed:
edited email subject line and message

Commit:
A minor error fix

Thanks @mdschnei for the headsup!

miralsumrani committed:
Update index.js

miralsumrani committed:
Update index.js

miralsumrani committed:
Update client.js

miralsumrani committed:
Update client.js

miralsumrani committed:
Update index.js

Commit:
Minor fixes!

Nicole changed the database and I had to change the natural join to left join.

Alikpld committed:
Update index.js

Minor changes to the query to filter by User_id

Commit:
Minor fixes!

Nicole made some changes on the database so I had to change natural join to left join.

Alikpld committed:
Update client.js

minor fix on closing the comment section

hibatawfeeq committed:
Merge pull request #27 from nmdoran/sortbytype

Filter by Food Type

nandatanvi committed:
Update email.js

mdschnei committed:
Merge branch 'main' of https://github.com/nmdoran/MSCI342 into addCustom

hibatawfeeq committed:
Merge branch 'main' into Integrate-search-with-add-product

hibatawfeeq committed:
Merge pull request #28 from nmdoran/Integrate-search-with-add-product

Integrate Search with Add Product

Alikpld committed:
Merge branch 'main' into Connect-search-to-main-page

nandatanvi committed:
Update email.js

Alikpld committed:
Merge pull request #29 from nmdoran/updateDatabase

Updated Database

nandatanvi committed:
Update email.js

email is set to send to Matthew at 12:30pm tomorrow for demo purposes.

nandatanvi committed:
Merge pull request #30 from nmdoran/addCustom

Add custom

Alikpld committed:
Merge branch 'main' into Connect-search-to-main-page

Alikpld committed:
Merge branch 'main' into Connect-search-to-main-page

Alikpld committed:
Merge pull request #26 from nmdoran/Connect-search-to-main-page

Connected the search function to the main page

mdschnei committed:
Merge branch 'main' of https://github.com/nmdoran/MSCI342 into editQuantity

mdschnei committed:
Merge branch 'main' of https://github.com/nmdoran/MSCI342 into editQuantity

nmdoran committed:
Merge pull request #25 from nmdoran/editQuantity

Edit quantity function

nmdoran committed:
Merge pull request #31 from nmdoran/email-functionality-trigger

edited email subject line and message

mdschnei committed:
Merge branch 'sortbyexpdate2' of https://github.com/nmdoran/MSCI342 into sortbyexpdate2
